dbt/adapters/redshift/connections.py

```diff
@@ -78,13 +78,30 @@ def _connection_keys(self):
             "cluster_id",
             "iam_profile",
             "sslmode",
+            "region",
         )
 
     @property
     def unique_field(self) -> str:
         return self.host
 
 
+def _get_aws_regions():
+    url = "https://ip-ranges.amazonaws.com/ip-ranges.json"
+    response = urllib.request.urlopen(url)
+    data = json.loads(response.read().decode())
+    regions = set()
+
+    for prefix in data["prefixes"]:
```

Comment:
These guaranteed to exist? Asking in case we run into a dreference exception that  handled with a `get`

Reply:
Yeah, I wanted a nicer way to do this but couldn't find a utility in the redshift connector and something public.
Separately, adding any common regions would need to include US, EU, APAC at least and the list is long.
I didn't want to add static variables that folks have to append when AWS adds a new region.
@sathiish-kumar any suggestions here?

Reply:
If it does turn out this is the best way to do it – which I don't think it's a problem – I would just recommend handling edge cases where this list might come back null or `prefix`/its elements comes back null. If we don't think that would ever happen (valid), we've at least documented here in the discussion that we thought about that

Reply:
@VersusFacit: How about this: [skip validation](https://github.com/dbt-labs/dbt-redshift/pull/420/commits/b598e79ffab9bfc9c98aeb76acfd6f1a47984ff3)?

Reply:
Looks good to me – thanks for taking the extra time to fill that out just in case. I feel like I've been addressing several bugs of this nature last mosth so I just thought to call it out

Reply:
@nssalian the other way would be to leverage boto3 - something along the lines of

```
client = boto3.client('redshift')
available_regions = [region['RegionName'] for region in client.describe_regions()['Regions']]
```

Reply:
the file doesn't have an explicit boto import already and boto clients need local AWS creds to resolve so that could fail if the env isn't set up correctly.

Reply:
Sounds fair @nssalian - the other concern is around the 7k+ prefixes that need to be parsed each time, we could try caching it in the future. As mentioned offline, we could use a top-level variable in the file to cache this and then keep referencing it, we could keep it as a constructor side-effect as well but that would not be my first choice.